File: DSA/Cpp/src/dsa/nodes.cpp
#include "dsa/nodes.h"

namespace dsa
{
    namespace nodes
    {
        Node::Node() : val(0) {}
        Node::Node(int val) : val(val) {}

        ListNode::ListNode() : Node(0), next(nullptr) {}
        ListNode::ListNode(int val) : Node(val), next(nullptr) {}
        ListNode::ListNode(int val, ListNode *next) : Node(val), next(nullptr) {}

        TreeNode::TreeNode() : Node() {}
        TreeNode::TreeNode(int val) : Node(val) {}

        BinaryTreeNode::BinaryTreeNode() : TreeNode() {}
        BinaryTreeNode::BinaryTreeNode(int val) : TreeNode(val) {}
        BinaryTreeNode::BinaryTreeNode(int val, BinaryTreeNode *left, BinaryTreeNode *right)
        {
            this->val = val;
            this->left = left;
            this->right = right;
        }

    }; // namespace nodes
} // namespace dsa

File: DSA/Python/src/dsa/nodes.py
class Node:
    def __init__(self, val: int = 0):
        self.val = val


class ListNode(Node):
    def __init__(self, val: int = 0, next=None):
        super().__init__(val)
        self.next = next


class TreeNode(Node):
    def __init__(self, val: int = 0):
        super().__init__(val)


class BinaryTreeNode(TreeNode):
    def __init__(self, val: int = 0, left=None, right=None):
        super().__init__(val)
        self.left = left
        self.right = right
